core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -1190,11 +1195,18 @@ public boolean isView(Session session, QualifiedObjectName viewName)
 
         Optional<String> owner = connectorView.get().getOwner();
         Identity runAsIdentity = systemSecurityMetadata.getViewRunAsIdentity(session, viewName.asCatalogSchemaTableName(), owner)
-                .or(() -> owner.map(Identity::ofUser))
+                .or(() -> owner.map(this::createViewOwnerIdentity))
                 .orElseThrow(() -> new TrinoException(NOT_SUPPORTED, "Catalog does not support run-as DEFINER views: " + viewName));
```

Comment:
Just a note that this error message doesn't make sense to me. The view is `SECURITY DEFINER` and there's no owner - looks more like metadata inconsistency?

Reply:
👍